src/web/PopupContainer.tsx

```diff
@@ -0,0 +1,106 @@
+/**
```

Comment:
How different is this from the native implementation? If it's very close, we should collapse the two into a common file rather than duplicating the code.

Reply:
They are very different. Perhaps some of the positioning logic in web/RootView.tsx could be moved to web/PopupContainer.tsx and then a subset of it extracted to a common base but that feels like a separate piece of work.